Refactor Java Service Certs and keys generation with DER and generate multiple crypt4gh keys

blankdots opened the pull request:
<!-- Lines like this one are comments and will not be shown in the final output. -->
<!-- Make sure that you have read the "Contributing" section of the README and also the notes in CodingStyle. -->
<!-- If you are a collaborator, please add labels and assign other collaborators for a review. -->

### Describe the pull request:
<!-- Replace [ ] with [x] to select options. -->
- [x] Functional change
- [x] Documentation change
### Pull request long description:
<!-- Describe your pull request in detail. -->

x509 certifiactes for the java services should be in DER format and added generation of 2 crypt4gh keys and optionally the PGP key.

### Changes made:
<!-- Enumerate the changes with 1), 2), 3) etc. -->
<!-- Ensure the test cases are updated if needed. -->
1. added option to specify from a JSON file the java services
2. `.crt` and `.key` are also encoded in DER format
3. generate 2 crypt4GH keys one for EGA (recipient) and one for a user e.g. Scientist
  - `ega_key.c4gh.sec` and  `ega_key.c4gh.pub` for EGA with password `ega_c4gh_passphrase` in `trace.yml`
  - `user_key.c4gh.sec` and  `user_key.c4gh.pub` for user with password `user_c4gh_passphrase` in `trace.yml`
4. Updated documentation
5. bumped to `0.4.0`

### Related issues:
<!-- Reference issues with #<issue-num>. -->
<!-- Write "Fixes #<issue-num>" to notify Github that this PR fixes an issue. -->

Fixes #36 

### Additional information:
<!-- Include sample dive log or other relevant information to allow testing the change where feasible. -->
Needs some testing one can also change the keystore using the `--java-store` and the password with `--java-store-pass`


### Documentation change:
<!-- If this PR makes changes to user functionality, then the documentation has to be updated too. -->
<!-- Please, briefly outline here what has changed in terms of the user experience (UX). -->
<!-- If UX changes have been made, a maintainer should apply the 'needs-documentation-change' label. -->

Updated missing documentation for crypt4gh update and default to JKS keystore

jbygdell reviewed (dismissed):
This will require a bit more changes than this. It's the format of the client key that needs to be in DER format. The cryptography package can produce PEM and DER formats. So it would be better to create the .cry .key files for the Java services in DER rather than PEM.

blankdots commented:
It could be a argument to the class `SecurityConfigGenerator` that will use either `serialization.Encoding.PEM` or `serialization.Encoding.DER` depending on the service. Would that work ?

jbygdell commented:
That would work, just set the default for Java services to DER.

blankdots committed:
read Java-based services from cli in a JSON file

Thist provides more flexibility to selecting Java based services.

blankdots committed:
Generate DER Certs and keys for Java Services

and it will generate DER certificates and Keys for each of them

blankdots committed:
bump version to 4.0

blankdots committed:
generate EGA (recipient) and user key for crypt4gh

We generate two keys for crypt4gh recipient and user, as well as adding the option to opt out of PGP key generation.
User key is optional as one key can be used for test purposes.

blankdots committed:
update documentation to reflect changes
